internal/controller/controllers/clusterdeployments_controller_test.go

```diff
@@ -3507,12 +3562,13 @@ var _ = Describe("unbindAgents", func() {
 		mockInstallerInternal = bminventory.NewMockInstallerInternals(mockCtrl)
 		mockClusterApi := cluster.NewMockAPI(mockCtrl)
 		cr = &ClusterDeploymentsReconciler{
-			Client:     c,
-			APIReader:  c,
-			Scheme:     scheme.Scheme,
-			Log:        common.GetTestLog(),
-			Installer:  mockInstallerInternal,
-			ClusterApi: mockClusterApi,
+			Client:           c,
+			APIReader:        c,
+			Scheme:           scheme.Scheme,
+			Log:              common.GetTestLog(),
+			Installer:        mockInstallerInternal,
+			ClusterApi:       mockClusterApi,
+			PullSecretClient: NewPullSecretHandler(c, c, mockInstallerInternal),
 		}
 
```

Comment:
Can we add a test for the invalid secret flow.

Reply:
I will